packages/api/src/controllers/stripe.test.ts

```diff
@@ -0,0 +1,158 @@
+import serverPromise, { TestServer } from "../test-server";
+import { clearDatabase } from "../test-helpers";
+import { ObjectStore, User } from "../schema/types";
+import { db } from "../store";
+import { v4 as uuid } from "uuid";
+import {
+  addDefaultProjectId,
+  deleteCredentials,
+  getS3PresignedUrl,
+  toObjectStoreUrl,
+  toWeb3StorageUrl,
+} from "./helpers";
+import { Request, Response } from "express";
+import { calculateOverageOnMinimumSpend } from "./stripe";
+
+let server: TestServer;
+
+// jest.setTimeout(70000)
+
+beforeAll(async () => {
+  server = await serverPromise;
+});
+
+afterEach(async () => {
+  await clearDatabase(server);
+});
+
+describe("controllers/stripe", () => {
+  describe("calculate usage", () => {
+    it("should calculate correct overage on minimum spend plans", async () => {
+      const product = {
+        minimumSpend: true,
+        monthlyPrice: 100,
+        usage: [
+          {
+            name: "Transcoding",
+            description: "Transcoding (minutes)",
+            price: 0.0055,
+            limit: 3_000,
+          },
+          {
+            name: "Delivery",
+            description: "Delivery (minutes)",
+            price: 0.0005,
+            limit: 100_000,
+          },
+          {
+            name: "Storage",
+            description: "Storage (minutes)",
+            price: 0.0015,
+            limit: 10_000,
+          },
+        ],
+      };
+
+      let billingUsage = {
```

Comment:
Can you add a little comment before each of these explaining what you are testing? I don't get why everyting is 0 here for example. I'll exemplify with some hypothesis I have for this one:

```suggestion
      // no charges when all usages are below the minimum spend
      let billingUsage = {
```
(but then no idea why the one below also has everything 0. need to clarify on each comment)

Another way of organizing this would be making each of these a separate `it()` test and you describe what you are testing in the test name instead of a comment (`product` can be a static const on top of file). That would also be easier to inspect and debug when running the tests, but just some comments would already be enough here.

Reply:
Added some comments for each step. Won't put it right now in separate `it()`, I would like to write all tests for billing here in billing v2 - so I will refactor it